ArchiSteamFarm/Bot.cs

```diff
@@ -2836,6 +2835,232 @@ public sealed class Bot : IAsyncDisposable {
+					.Where(item => item.Tradable)
+					.Where(item => (item.Type == Steam.Asset.EType.TradingCard) || (item.Type == Steam.Asset.EType.FoilTradingCard))
+					.Where(item => appIDs.Contains(item.RealAppID))
+					.ToHashSetAsync()
+					.ConfigureAwait(false);
+			} catch (HttpRequestException e) {
+				ArchiLogger.LogGenericWarningException(e);
+
+				return;
+			} catch (Exception e) {
+				ArchiLogger.LogGenericException(e);
+
+				return;
+			}
+
+			if (inventory.Count == 0) {
+				ArchiLogger.LogGenericWarning(string.Format(Strings.ErrorIsEmpty), nameof(inventory));
+
+				return;
+			}
+
+			HashSet<Steam.Asset> result = new HashSet<Steam.Asset>();
+
+			foreach (uint appID in appIDs) {
+				HashSet<Steam.Asset> itemsForAppBadges = await GetItemsForFullBadge(inventory, appID).ConfigureAwait(false);
+
+				result.UnionWith(itemsForAppBadges);
+			}
+
+			if (result.Count > 0) {
+				await Actions.SendInventory(inventory: result).ConfigureAwait(false);
+			}
+		}
+
+		private async Task<HashSet<Steam.Asset>> GetItemsForFullBadge(IReadOnlyCollection<Steam.Asset> inventory, uint appID) {
+			byte requiredCards = await ArchiWebHandler.GetCardCountForGame(appID).ConfigureAwait(false);
+
+			Dictionary<Steam.Asset.EType, HashSet<Steam.Asset>> inventoryForAppIDPerType = inventory.Where(item => item.RealAppID == appID).GroupBy(item => item.Type).ToDictionary(grouping => grouping.Key, grouping => grouping.ToHashSet());
+
+			HashSet<Steam.Asset> result = new HashSet<Steam.Asset>(0);
+
+			if (inventoryForAppIDPerType.TryGetValue(Steam.Asset.EType.TradingCard, out HashSet<Steam.Asset>? availableTradingCards)) {
+				result.UnionWith(GetItemsForFullBadge(availableTradingCards, requiredCards));
+			}
+
+			if (inventoryForAppIDPerType.TryGetValue(Steam.Asset.EType.FoilTradingCard, out HashSet<Steam.Asset>? availableFoilCards)) {
+				result.UnionWith(GetItemsForFullBadge(availableFoilCards, requiredCards));
+			}
+
+			return result;
+		}
+
+		private static HashSet<Steam.Asset> GetItemsForFullBadge(IReadOnlyCollection<Steam.Asset> availableItems, byte cardsPerBadge) {
+			Dictionary<ulong, List<Steam.Asset>> itemsPerClassId = availableItems.GroupBy(item => item.ClassID).ToDictionary(grouping => grouping.Key, grouping => grouping.OrderByDescending(item => item.Amount).ToList());
+
+			if (itemsPerClassId.Keys.Count != cardsPerBadge) {
+				// This can happen if cards are not tradable or we traded away a card between checking which badges are craftable and what our inventory state is
+				return new HashSet<Steam.Asset>(0);
+			}
+
+			// "maximum" as it could happen, that we need to send less sets than exist in our inventory. This only is possible if Valve ever decides to send us Assets with other amount than 1
+			// example: We have a set that is craftable 7 times. In our inventory we have item with class id 0 and amount 5 twice  and class id 1 4 times with amount 2. All other class ids are available seven times with an amount of 1 each.
```

Comment:
As far as I know, Steam allows to split item stack when trading item, e.g. if you have 1000 gems, you can send in the trade 500 of them and 500 will stay in the inventory, so we are still available to send all available sets.

Reply:
Can't currently test if (and how) this works with trading cards. I think it's better to leave it as is. If you are sure how it works and you know it will work with trading cards as well, you are welcome to change the code.

Reply:
It works with any stackable items (stackable item = item with amount > 1, ref: [economy.js](https://steamcommunity-a.akamaihd.net/public/javascript/economy.js?v=rI0xgpCYSBS9&l=english), line 394), ref: [economy_trade.js](https://steamcommunity-a.akamaihd.net/public/javascript/economy_trade.js?v=_S8IrXmFV6F8&l=english), function MoveItemToTrade

Reply:
Neat. Do you also know how we can send other quantities in ASF code? I found nothing (and I think just copying all properties and changing amount will not work [it would match usual Valve logic tho])

Reply:
I guess we can use Asset constructor to create asset with needed amount, AFAIK all the properties should be the same except for the amount.

Reply:
Does this mean we can send one item with amount 5 and steam will combine all our items of that type into one if the amount of every single one is lower or does it only work when decreasing the amount?

Reply:
No, it only works when we send a part of the stacked item (decreasing the amount), Steam doesn't give an option to combine items into stack in the trade.

Reply:
Should be working in latest commit. Please take a look and tell me what you think